_includes/v20.1/app/gorm-sample.go

```diff
@@ -146,7 +146,13 @@ func runTransaction(db *gorm.DB, fn txnFunc) error {
 			// We need to cast GORM's db.Error to *pq.Error so we can
 			// detect the Postgres transaction retry error code and
 			// handle retries appropriately.
-			pqErr := err.(*pq.Error)
+			pqErr, ok := err.(*pq.Error)
+			if !ok || pqErr.Code != "40001" {
+				// If it's not a retry error, it's some other sort of
+				// DB interaction error that needs to be handled by
+				// the caller.
+				return err
+			}
 			if pqErr.Code == "40001" {
```

Comment:
I don't think this conditional is useful as the previous one already catches all other cases.

Reply:
You are right, I should remove that wrapping